--- mcomix-1.2.1-master/mcomix/main.py	2018-06-26 20:40:01.027742648 -0700
+++ mcomix-1.2.1/mcomix/main.py	2018-10-09 18:37:40.379431149 -0700
@@ -36,7 +36,8 @@
 from mcomix import log
 import math
 import operator
-
+import shutil
+import errno
 
 class MainWindow(gtk.Window):
 
@@ -1019,6 +1020,47 @@
 
         save_dialog.destroy()
 
+    def move_file(self, *args):
+        current_file = self.imagehandler.get_real_path()
+        dir_target = os.path.join(os.path.dirname(current_file), 'keep')
+        file_target = os.path.join(dir_target, os.path.basename(current_file))
+
+        if not os.path.exists(dir_target):
+            try:
+                os.makedirs(dir_target)
+            except OSError as exc: # Guard against race condition
+                if exc.errno != errno.EEXIST:
+                   raise
+
+        if self.filehandler.archive_type is not None:
+            self.filehandler.last_read_page.clear_page(current_file)
+
+            next_opened = self.filehandler._open_next_archive()
+            if not next_opened:
+                next_opened = self.filehandler._open_previous_archive()
+            if not next_opened:
+                self.filehandler.close_file()
+
+            if os.path.isfile(current_file):
+                os.rename(current_file, file_target)
+        else:
+            if self.imagehandler.get_number_of_pages() > 1:
+                # Open the next/previous file
+                if self.imagehandler.get_current_page() >= self.imagehandler.get_number_of_pages():
+                    self.flip_page(-1)
+                else:
+                    self.flip_page(+1)
+                # Move the desired file
+                if os.path.isfile(current_file):
+                    os.rename(current_file, file_target)
+
+                # Refresh the directory
+                self.filehandler.refresh_file()
+            else:
+                self.filehandler.close_file()
+                if os.path.isfile(current_file):
+                    os.rename(current_file, file_target)
+
     def delete(self, *args):
         """ The currently opened file/archive will be deleted after showing
         a confirmation dialog. """
